### ISSUE TITLE
[Bug]: Reader follow button sometimes follows the wrong site

### ISSUE THREAD

**Issue by creativecoder** [Automattic/wp-calypso#69207]
### Quick summary

Clicking follow in reader sometimes follows the wrong site.

The problem will be the same with anything that uses the POST https://public-api.wordpress.com/rest/v1.1/read/following/mine/new endpoint.

### Steps to reproduce

1. Go to https://wordpress.com/read/feeds/70239806
2. Open the browser dev tools and watch the network tab
3. Click the Follow button

### What you expected to happen

I follow the feed https://specifierreview.com/

### What actually happened

- I follow the site kirhammond.wordpress.com
- When I refresh the page for https://wordpress.com/read/feeds/70239806, the follow button indicates I'm not following the site

### Browser

Mozilla Firefox

### Context

Reader testing

### Platform (Simple, Atomic, or both?)

Atomic, Self-hosted

### Other notes

<img width="484" alt="image" src="https://user-images.githubusercontent.com/1699996/196484928-801d2290-3c5a-404a-8297-d39e633b095d.png">

### Reproducibility

Intermittent, only for some feeds/sites

### Severity

Some (< 50%)

### Available workarounds?

No but the platform is still usable

**Comment by creativecoder** [comment]
Note that this problem seems to be caused by how we prioritize using domain mappings of Simple sites, when searching for a site by url.

In the specific case of https://wordpress.com/read/feeds/70239806, the site kirhammond.wordpress.com looks like it has a domain mapping for specifierreview.com so ends up being the site followed. specifierreview.com looks to be hosted outside of WP.com.

**Comment by creativecoder** [comment]
As a stopgap, maybe there's a way to follow the feed directly via the API, rather than using `https://public-api.wordpress.com/rest/v1.1/read/following/mine/new` and specifying the URL (which is what the follow button is doing now)

**Comment by roo2** [comment]
I investigate this also. I tracked it through to`global_get_blog_info` 2e450-pb/ which is called with args `('specifierreview.com', '/')`  . `global_get_blog_info` then calls `get_blog_info` in 2e451-pb, and goes through the domain mapping check and finds `kirhammond.wordpress.com`. So that's the site that gets used for the subscription.

I actually found that there is a commented out line of code here 2e452-pb that explains: 

`// Check if this blog is domain mapped outside of WordPress.com`

The next line is a commented out SQL query that selects domain mapping with type of `DOMAIN_REDIRECTED`.

It almost looks to me like someone started to address this case but then gave up? I traced the git blame but couldn't get any more information on the commented out line.

If I'm understanding this correctly, I think we can resurect that commented out sql query and ignore $blog if it is set up as a domain redirect. This would allow the subsequent code at to identify the external site correctly 2e45a-pb.

I tested this and it worked for the one test site, we just have to make sure it wouldn't adversely impact any other sites with different configurations. I'll make a diff
